File: solutions/728.self-dividing-numbers/iteration.ts
/*
 * @lc app=leetcode.cn id=728 lang=typescript
 *
 * https://leetcode-cn.com/problems/self-dividing-numbers
 */

// @lc code=start
function isSelfDividing(num: number): boolean {
    let quotient = num;
    while (quotient > 0) {
        const digit = quotient % 10;
        if (digit === 0 || num % digit !== 0) {
            return false;
        }
        quotient = Math.floor(quotient / 10);
    }
    return true;
}

function selfDividingNumbers(left: number, right: number): number[] {
    const result: number[] = [];
    for (let i = left; i <= right; i++) {
        if (isSelfDividing(i)) {
            result.push(i);
        }
    }
    return result;
}
// @lc code=end

export default selfDividingNumbers;

File: solutions/78.subsets/src/iteration.rs
/*
 * @lc app=leetcode.cn id=78 lang=rust
 *
 * https://leetcode.cn/problems/subsets/
 */

pub struct Solution;

// @lc code=start
impl Solution {
    pub fn subsets(nums: Vec<i32>) -> Vec<Vec<i32>> {
        let mut result = vec![];
        for mask in 0..1 << nums.len() {
            let mut sub = vec![];
            for i in 0..nums.len() {
                if mask & (1 << i) == 0 {
                    continue;
                }
                sub.push(nums[i]);
            }
            result.push(sub);
        }
        result
    }
}
// @lc code=end
